src/main/java/com/google/sps/data/SessionContext.java

```diff
@@ -19,7 +21,18 @@ public SessionContext(UserRepository userRepository) {
     userService = UserServiceFactory.getUserService();
     this.userRepository = userRepository;
   }
+  /**
+  * Getter method for getting the instance
+  */
 
+  public static SessionContext getInstance() {
+      if(instance == null){
+           UserRepository userRepo = new PersistentUserRepository();
```

Comment:
This can be `UserRepository userRepo = PersistentUserRepository.getInstance();`

Reply:
Oh right. Good suggestion